File: Berita-Indo-Java/app/src/main/java/com/hafizhmo/beritaindojava/data/News.java
package com.hafizhmo.beritaindojava.data;
import com.google.gson.annotations.SerializedName;

import java.util.List;

public class News {

    //TO-D0 1: initial attribute 'title', 'puDate', 'thumb' and 'link' using SerializedName
    @SerializedName("data")
    public Data data;

    public static class Data {
        @SerializedName("posts")
        public List<Post> posts;

        public static class Post {
            @SerializedName("title")
            public String title;
            @SerializedName("pubDate")
            public String pubDate;
            @SerializedName("thumbnail")
            public String thumbnail;
            @SerializedName("link")
            public String link;
        }
    }
}

File: Berita-Indo-Kotlin/app/src/main/java/com/hafizhmo/beritaindokotlin/data/News.kt
package com.hafizhmo.beritaindokotlin.data

import android.os.Parcelable
import com.google.gson.annotations.SerializedName
import kotlinx.android.parcel.Parcelize

@Parcelize
data class News(
    @SerializedName("data")
    val datas: Data
) : Parcelable {

    @Parcelize
    data class Data(
        @SerializedName("posts")
        val posts: List<Post>
    ) : Parcelable {

        @Parcelize
        data class Post(
            @SerializedName("title")
            val title: String?,
            @SerializedName("pubDate")
            val pubDate: String?,
            @SerializedName("description")
            val description: String?,
            @SerializedName("thumbnail")
            val thumbnail: String?,
            @SerializedName("link")
            val link: String?
        ) : Parcelable
    }
}


